server/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java

```diff
@@ -94,14 +97,17 @@ public AggregatedDfs aggregateDfs(Collection<DfsSearchResult> results) {
             assert terms.length == stats.length;
             for (int i = 0; i < terms.length; i++) {
                 assert terms[i] != null;
+                if (stats[i] == null) {
+                    continue;
+                }
                 TermStatistics existing = termStatistics.get(terms[i]);
                 if (existing != null) {
                     assert terms[i].bytes().equals(existing.term());
                     // totalTermFrequency is an optional statistic we need to check if either one or both
                     // are set to -1 which means not present and then set it globally to -1
                     termStatistics.put(terms[i], new TermStatistics(existing.term(),
-                            existing.docFreq() + stats[i].docFreq(),
-                            optionalSum(existing.totalTermFreq(), stats[i].totalTermFreq())));
+                        existing.docFreq() + stats[i].docFreq(),
+                        optionalSum(existing.totalTermFreq(), stats[i].totalTermFreq())));
```

Comment:
I think we don't need the sum to be optional anymore since ttf may no longer be -1?

Reply:
Right, I pushed https://github.com/elastic/elasticsearch/pull/33310/commits/647b846d21c55a72dae0a512ada1dde15ecbfa26